iwyu_location_util.cc

```diff
@@ -142,6 +143,10 @@ SourceLocation GetLocation(const clang::Stmt* stmt) {
   } else if (const ConditionalOperator* conditional_op =
              DynCastFrom(stmt)) {
     return conditional_op->getQuestionLoc();
+  } else if (const UnaryOperator* unary_op = DynCastFrom(stmt)) {
+    // Drill through unary operators and parentheses, to get at the underlying
+    // DeclRefExpr or whatever, e.g. '*(x)' should give the location of 'x'
+    stmt = unary_op->getSubExpr()->IgnoreParenImpCasts();
```

Comment:
Nice how you are handling unary operator location. It is better than lexing macros.